tests/unit/src/Internal/Service/PaymentProcessingServiceTest.php

```diff
@@ -71,7 +79,7 @@ public function test_process_payment_happy_path() {
 		$mock_initial_state   = $this->createMock( InitialState::class );
 		$mock_completed_state = $this->createMock( CompletedState::class );
 
-		$sut = new PaymentProcessingService( $this->mock_state_factory, $this->mock_legacy_proxy );
+		$sut = new PaymentProcessingService( $this->mock_state_factory, $this->mock_legacy_proxy, $this->mock_context_logger );
```

Comment:
We're adding the mock here, but not using it. Let's make sure that `log_changes` is called.

Reply:
Done